### DISCUSSION TITLE
Failing to instantiate a `Store`

### DISCUSSION THREAD

**Discussion by appfrosch** [pointfreeco/swift-composable-architecture#2337] [Q&A] [answered]
I just upgraded a little project of mine to TCA 1.0.0 and now am having the issue of not being able to instantiate a `Store` any more.
I am on Xcode 15 beta 5.

I cannot see where a possible mistake of mine could be the root cause, but maybe I am just not seeing it?

I created a minimal Reducer to show the issue:
```swift
import ComposableArchitecture
import SwiftUI

struct SampleFeature: Reducer {
    struct State: Equatable {}
    enum Action {}
    var body: some ReducerOf<Self> {
        Reduce { state, action in
            return .none
        }
    }
}

struct SampleFeatureView: View {
    let store: StoreOf<SampleFeature>
    
    var body: some View {
        WithViewStore(store, observe: { $0 }) { viewStore in
            Text("Hello, World!")
        }
    }
}

#Preview {
    SampleFeatureView(
        store: Store(
            initialState: SampleFeature.State(),
            reducer: SampleFeature()
        )
    )
}
```

The errors Xcode is showing me down in the preview are:
- `Generic parameter 'R' could not be inferred`
- `Cannot convert value of type 'SampleFeature.State' to expected argument type '_'`
- Cannot convert value of type 'SampleFeature' to expected argument type '() -> R'

![Xcode-ProjectTracker — SampleFeature swift — Edited-20230731-094917](https://github.com/pointfreeco/swift-composable-architecture/assets/91844195/b7dae1ac-976e-41ec-a586-c6e2a6a2c6fb)

**Comment by Alex293** [comment, marked as answer]
Store now take a closure for the reducer arg like: `reducer: { SampleFeature() }` ; have a look at the third error message

**Comment by appfrosch** [comment]
Oh, I totally overlooked this (even though watching  [episode 243](https://www.pointfree.co/episodes/ep243-tour-of-the-composable-architecture-1-0-the-basics) earlier).

I can totally confirm that 
```swift
#Preview {
    SampleFeatureView(
        store: Store(
            initialState: SampleFeature.State()) {
                SampleFeature()
            }
    )
}
```
works. 

I am also realizing that I have to improve reading (and understanding …) these error messages–with your much appreciated (and blazingly fast 🤯) reply I do understand, but I failed seeing it myself...

Thanks for your help!

**Reply by Alex293** [reply]
As the generics are deduced by the compiler, the error messages can be quite confusing. You can often simply specify the missing types or generics to get precise errors. Here specifying the generics for the Store would yield a single error about the reducer.

**Comment by appfrosch** [comment]
You mean something like this:
```swift
let store = Store<ProjectListFeature.State, ProjectListFeature.Action>( … )
```
In this specific case, the error messages remain the same...
